contract/src/modalities.rs

```diff
@@ -210,6 +210,15 @@ impl TokenIdentifier {
     }
 }
 
+impl ToString for TokenIdentifier {
+    fn to_string(&self) -> String {
+        match self {
+            TokenIdentifier::Index(i) => i.to_string(),
+            TokenIdentifier::Hash(h) => h.to_string(),
```

Comment:
Please avoid naming vars with one car

```suggestion
            TokenIdentifier::Index(index) => index.to_string(),
            TokenIdentifier::Hash(hash) => hash.to_string(),
```

Reply:
Applied suggestion